content/en/platform/corda/4.12/enterprise/upgrade-guide.md

```diff
@@ -0,0 +1,265 @@
+
+## Starting 4.12 nodes
+
+1. Run the database migration scripts.
+2. Start your node in the usual way:
+
+   ```
+   java -jar corda-4.12.jar -f node.conf
+   ```
+
+## Adding new 4.12 nodes
+
+If you want to add another Corda 4.12 node into your network post-upgrade, the specific steps vary depending on the state of your network.
+
+### Adding new 4.12 nodes to a mixed network
+
+If you are operating a mixed network, then the process for adding a new Corda 4.12 node is relatively straightforward.
+
+1. Set up the node folder structure same as other Corda 4.12 nodes.
+2. Add the `legacy-contracts` folder and associated files to the node folder.
+
+   ```
+   .
+   ├── certificates
+   ├── corda-4.12.jar
+   ├── cordapps
+   │   ├── config
+   │   ├── corda-finance-contracts-4.12.jar
+   │   └── corda-finance-workflows-4.12.jar
+   ├── drivers
+   ├── legacy-contracts
+   │   └── corda-finance-contracts-4.11.jar
+   └── node.conf
+   ```
+
+3. Register the node to the network and proceed to operate it as normal.
+
+### Adding new 4.12 nodes to a non-mixed network
+
+Perform the following steps if you are adding a new Corda 4.12 node to a network that is only running Corda 4.12 nodes, but also contains older Corda 4.x transactions.
+
+{{< note >}}
+In this scenario, you must keep a copy of the old CorDapp contract JAR file.
+{{< /note >}}
+
+1. Set up the node folder structure without the `legacy-contracts` folder.
+
+   ```
+   .
+   ├── certificates
+   ├── corda-4.12.jar
+   ├── cordapps
+   │   ├── config
+   │   ├── corda-finance-contracts-4.12.jar
+   │   └── corda-finance-workflows-4.12.jar
+   ├── drivers
+   └── node.conf
+   ```
+
+2. Register and then start the node.
+
+3. Access the Corda node either via RPC client or the standalone shell and upload the old CorDapp contract JAR as an attachment to the node.
+   For more information on uploading attachments, see [Working with attachments]({{< relref "get-started/tutorials/supplementary-tutorials/tutorial-attachments.md" >}}).
+
+## Additional release information
+
+In Corda 4.12, the Corda node explorer is no longer included in the release pack. Should you require it, the Corda 4.11 node explorer remains compatible with Corda 4.12 and should be used for Corda 4.12 nodes.
+
+Corda 4.11 versions of the node and flow management plugins are also compatible with Corda 4.12 and you should use them for Corda 4.12 nodes.
+
+## Troubleshooting
+
+### Add-opens
+
+In Java 17, an application can no longer access an internal Java method due to the new module system. If a package is not opened, access will be denied. You can circumvent this strong encapsulation by either:
```

Comment:
I think this sounds better:

Since upgrading Corda to Java 17, it is important to keep in mind that any CorDapp will no longer have access to internal Java Classes due to the new strong Encapsulation module system introduced in Java 11.

If your CorDapp still requires access to any internal Java classes you can open the package either by: